src/cursor.rs

```diff
@@ -238,32 +236,24 @@ impl Cursor {
         // Send read_preference to the server based on the result from server selection.
         let new_query = if !send_read_pref {
             query
+        } else if query.get("$query").is_some() {
```

Comment:
Given that the pattern match wasn't actually capturing any variables, it seemed more eadable to change it to a pair of `else (if)` clauses.